opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -220,42 +220,97 @@ message Span {
   // enforced. If this value is 0, then no links were dropped.
   uint32 dropped_links_count = 14;
 
-  // An optional final status for this span. Semantically when Status
-  // wasn't set it is means span ended without errors and assume
-  // Status.Ok (code = 0).
+  // An optional final status for this span. Semantically when Status isn't set it means
+  // span's status code is unset, i.e. assume STATUS_CANONICAL_CODE_UNSET (code = 0).
   Status status = 15;
 }
 
 // The Status type defines a logical error model that is suitable for different
 // programming environments, including REST APIs and RPC APIs.
 message Status {
 
-  // StatusCode mirrors the codes defined at
-  // https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#statuscanonicalcode
-  enum StatusCode {
-    STATUS_CODE_OK                  = 0;
-    STATUS_CODE_CANCELLED           = 1;
-    STATUS_CODE_UNKNOWN_ERROR       = 2;
-    STATUS_CODE_INVALID_ARGUMENT    = 3;
-    STATUS_CODE_DEADLINE_EXCEEDED   = 4;
-    STATUS_CODE_NOT_FOUND           = 5;
-    STATUS_CODE_ALREADY_EXISTS      = 6;
-    STATUS_CODE_PERMISSION_DENIED   = 7;
-    STATUS_CODE_RESOURCE_EXHAUSTED  = 8;
-    STATUS_CODE_FAILED_PRECONDITION = 9;
-    STATUS_CODE_ABORTED             = 10;
-    STATUS_CODE_OUT_OF_RANGE        = 11;
-    STATUS_CODE_UNIMPLEMENTED       = 12;
-    STATUS_CODE_INTERNAL_ERROR      = 13;
-    STATUS_CODE_UNAVAILABLE         = 14;
-    STATUS_CODE_DATA_LOSS           = 15;
-    STATUS_CODE_UNAUTHENTICATED     = 16;
+  enum DeprecatedStatusCode {
+    DEPRECATED_STATUS_CODE_OK                  = 0;
+    DEPRECATED_STATUS_CODE_CANCELLED           = 1;
+    DEPRECATED_STATUS_CODE_UNKNOWN_ERROR       = 2;
+    DEPRECATED_STATUS_CODE_INVALID_ARGUMENT    = 3;
+    DEPRECATED_STATUS_CODE_DEADLINE_EXCEEDED   = 4;
+    DEPRECATED_STATUS_CODE_NOT_FOUND           = 5;
+    DEPRECATED_STATUS_CODE_ALREADY_EXISTS      = 6;
+    DEPRECATED_STATUS_CODE_PERMISSION_DENIED   = 7;
+    DEPRECATED_STATUS_CODE_RESOURCE_EXHAUSTED  = 8;
+    DEPRECATED_STATUS_CODE_FAILED_PRECONDITION = 9;
+    DEPRECATED_STATUS_CODE_ABORTED             = 10;
+    DEPRECATED_STATUS_CODE_OUT_OF_RANGE        = 11;
+    DEPRECATED_STATUS_CODE_UNIMPLEMENTED       = 12;
+    DEPRECATED_STATUS_CODE_INTERNAL_ERROR      = 13;
+    DEPRECATED_STATUS_CODE_UNAVAILABLE         = 14;
+    DEPRECATED_STATUS_CODE_DATA_LOSS           = 15;
+    DEPRECATED_STATUS_CODE_UNAUTHENTICATED     = 16;
   };
 
-  // The status code. This is optional field. It is safe to assume 0 (OK)
-  // when not set.
-  StatusCode code = 1;
+  // The deprecated status code. This is an optional field.
+  //
+  // This field is deprecated and is replaced by `code` field below. See backward
+  // compatibility notes below. According to our stability guarantees this field
+  // will be removed in 12 months, on Sep 26, 2021. All usage of old senders and
```

Comment:
Not related to this PR but just noticed since it's defined very clearly here with a date - 12 months seems like a pretty short time for API stability post-GA.

Reply:
12 months is our formal definition of "stable": https://github.com/open-telemetry/community/blob/47813530864b9fe5a5146f466a58bd2bb94edc72/maturity-matrix.yaml#L57

Reply:
I think we need a new stability called "GA", that has stronger requirements.